bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/factory/SiemensHvacHandlerFactory.java

```diff
@@ -0,0 +1,110 @@
+    private final ChannelTypeRegistry channelTypeRegistry;
+    private final TranslationProvider translationProvider;
+
+    @Activate
+    public SiemensHvacHandlerFactory(final @Reference HttpClientFactory httpClientFactory,
+            final @Reference SiemensHvacMetadataRegistry metaDataRegistry,
+            final @Reference NetworkAddressService networkAddressService,
+            final @Reference ChannelTypeRegistry channelTypeRegistry,
+            final @Reference TimeZoneProvider timeZoneProvider,
+            final @Reference TranslationProvider translationProvider) {
+        this.httpClientFactory = httpClientFactory;
+        this.metaDataRegistry = metaDataRegistry;
+        this.networkAddressService = networkAddressService;
+        this.channelTypeRegistry = channelTypeRegistry;
+        this.translationProvider = translationProvider;
+
+        ConverterFactory.registerConverter(timeZoneProvider);
+    }
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SiemensHvacBindingConstants.BINDING_ID.equals(thingTypeUID.getBindingId());
+    }
+
+    @Override
+    public @Nullable Thing createThing(ThingTypeUID thingTypeUID, Configuration configuration,
+            @Nullable ThingUID thingUID, @Nullable ThingUID bridgeUID) {
+        if (SiemensHvacBindingConstants.THING_TYPE_OZW.equals(thingTypeUID)) {
+            ThingUID iPBridgeUID = getIPBridgeThingUID(thingTypeUID, thingUID, configuration);
+            return super.createThing(thingTypeUID, configuration, iPBridgeUID, null);
+        } else if (SiemensHvacBindingConstants.BINDING_ID.equals(thingTypeUID.getBindingId())) {
+            return super.createThing(thingTypeUID, configuration, thingUID, bridgeUID);
+        }
+        throw new IllegalArgumentException(
+                "The thing type " + thingTypeUID + " is not supported by the SiemensHvac binding.");
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        if (thing.getThingTypeUID().equals(SiemensHvacBindingConstants.THING_TYPE_OZW)) {
+            return new SiemensHvacBridgeThingHandler((Bridge) thing, networkAddressService, httpClientFactory,
+                    metaDataRegistry, translationProvider);
+        } else if (SiemensHvacBindingConstants.BINDING_ID.equals(thing.getThingTypeUID().getBindingId())) {
+            SiemensHvacHandlerImpl handler = new SiemensHvacHandlerImpl(thing,
+                    metaDataRegistry.getSiemensHvacConnector(), metaDataRegistry, channelTypeRegistry);
+            return handler;
+        }
+        return null;
+    }
+
+    private ThingUID getIPBridgeThingUID(ThingTypeUID thingTypeUID, @Nullable ThingUID thingUID,
+            Configuration configuration) {
+        if (thingUID != null) {
+            return thingUID;
+        }
+        String ipAddress = (String) configuration.get(SiemensHvacBindingConstants.IP_ADDRESS);
```

Comment:
This looks fishy, can you check https://github.com/openhab/openhab-addons/pull/14263#discussion_r1651557389?

Reply:
fix in next commit, it was useless and I removed it.